onnxruntime/core/graph/contrib_ops/shape_inference_functions.cc

```diff
@@ -8,7 +8,9 @@ namespace onnxruntime {
 namespace contrib {
 void EmbedLayerNormalizationShapeInference(::ONNX_NAMESPACE::InferenceContext& ctx) {
   propagateElemTypeFromInputToOutput(ctx, 2, 0);
-  propagateElemTypeFromInputToOutput(ctx, 0, 1);
+  if (ctx.getNumOutputs() > 1) {
```

Comment:
This is not correct. There are two optional outputs, so two outputs could be output + mask_index or output + embedding_sum.

Reply:
Updated this check to see if the mask input shape exists